contracts/defi/provider/CompoundV2Provider.sol

```diff
@@ -0,0 +1,405 @@
+     */
+    function mint(BaseWallet _wallet, address _cToken, address _token, uint256 _amount) internal {
+        require(_cToken != address(0), "Compound: No market for target token");
+        require(_amount > 0, "Compound: amount cannot be 0");
+        if(_token == ETH_TOKEN_ADDRESS) {
+            _wallet.invoke(_cToken, _amount, abi.encodeWithSignature("mint()"));
+        }
+        else {
+            _wallet.invoke(_token, 0, abi.encodeWithSignature("approve(address,uint256)", _cToken, _amount));
+            _wallet.invoke(_cToken, 0, abi.encodeWithSignature("mint(uint256)", _amount));
+        }
+    }
+
+    /**
+     * @dev Redeems underlying tokens from a cToken contract.
+     * @param _wallet The target wallet.
+     * @param _cToken The cToken contract.
+     * @param _amount The amount of cToken to redeem.
+     */
+    function redeem(BaseWallet _wallet, address _cToken, uint256 _amount) internal {     
+        require(_cToken != address(0), "Compound: No market for target token");   
+        require(_amount > 0, "Compound: amount cannot be 0");
+        _wallet.invoke(_cToken, 0, abi.encodeWithSignature("redeem(uint256)", _amount));
+    }
+
+    /**
+     * @dev Redeems underlying tokens from a cToken contract.
+     * @param _wallet The target wallet.
+     * @param _cToken The cToken contract.
+     * @param _amount The amount of underlying token to redeem.
+     */
+    function redeemUnderlying(BaseWallet _wallet, address _cToken, uint256 _amount) internal {     
+        require(_cToken != address(0), "Compound: No market for target token");   
+        require(_amount > 0, "Compound: amount cannot be 0");
+        _wallet.invoke(_cToken, 0, abi.encodeWithSignature("redeemUnderlying(uint256)", _amount));
+    }
+
+    /**
+     * @dev Borrows underlying tokens from a cToken contract.
+     * @param _wallet The target wallet.
+     * @param _cToken The cToken contract.
+     * @param _amount The amount of underlying tokens to borrow.
+     */
+    function borrow(BaseWallet _wallet, address _cToken, uint256 _amount) internal {
+        require(_cToken != address(0), "Compound: No market for target token");
+        require(_amount > 0, "Compound: amount cannot be 0");
+        _wallet.invoke(_cToken, 0, abi.encodeWithSignature("borrow(uint256)", _amount));
+    }
+
+    /**
+     * @dev Repays some borrowed underlying tokens to a cToken contract.
+     * @param _wallet The target wallet.
+     * @param _cToken The cToken contract.
+     * @param _amount The amount of underlying to repay.
+     */
+    function repayBorrow(BaseWallet _wallet, address _cToken, uint256 _amount) internal {
+        require(_cToken != address(0), "Compound: No market for target token");
+        require(_amount > 0, "Compound: amount cannot be 0");
+        string memory symbol = CToken(_cToken).symbol();
+        if(keccak256(abi.encodePacked(symbol)) == keccak256(abi.encodePacked("cETH"))) {
```

Comment:
Maybe it's simpler (and more consistent with what we do in `mint()`) to do:
```
address token = CToken(_cToken).underlying()
if(token == ETH_TOKEN_ADDRESS) { ... }
```

Reply:
Ah sorry, I just realised cETH does not implement `underlying()`, so ignore my comment